include/swift/AST/Types.h

```diff
@@ -3267,12 +3282,55 @@ class AnyFunctionType : public TypeBase {
   /// derivative function types, e.g. when type-checking `@differentiable` and
   /// `@derivative` attributes.
   AnyFunctionType *getAutoDiffDerivativeFunctionType(
-      IndexSubset *parameterIndices, unsigned resultIndex,
-      AutoDiffDerivativeFunctionKind kind,
+      IndexSubset *parameterIndices, AutoDiffDerivativeFunctionKind kind,
       LookupConformanceFn lookupConformance,
       GenericSignature derivativeGenericSignature = GenericSignature(),
       bool makeSelfParamFirst = false);
 
+  /// Returns the linear map function type returned by the derivative function
```

Comment:
The documentation seems unclear. Does this expect `this` to be "the derivative function", or any function? If you intend to use normal (original) functions as `this`, then the "returned by the derivative function" part is confusing -- the concept of "linear map functions of a function" doesn't have to be associated with "derivative functions". I'd just say "Returns the corresponding linear map function type for the given ...".

Reply:
Nice tip! Done:
```
  /// Returns the corresponding linear map function type for the given parameter
  /// indices, linear map function kind, and other auxiliary parameters.
```